src/ffi.rs

```diff
@@ -66,20 +46,20 @@ pub unsafe extern "C" fn newton_distribute_bodies(
     };
 
     let bodies = distributor.distribution();
-    let field = &mut *field;
-    field.bodies = bodies;
+    let environment = &mut *environment;
+    environment.bodies = bodies;
 }
 
 #[no_mangle]
-pub unsafe extern "C" fn newton_step(field: *mut Field) {
-    let field = &mut *field;
-    field.update()
+pub unsafe extern "C" fn newton_step(environment: *mut Environment) {
+    let environment = &mut *environment;
+    environment.update()
 }
 
 #[no_mangle]
-pub unsafe extern "C" fn newton_body_pos(field: *const Field, id: u32) -> NewtonPoint {
-    let field = &*field;
```

Comment:
it is a bit confusing here `&*enviroment` what does this little code mean?

Reply:
It's true, maybe there is a better way to write this. But basically we dereference `*const Environment` with `*environment`, then we reference it again with `&`.

Reply:
firstly dereference `*const Environment` with '*environment' and then reference it again with  `&` why do we have to do that? and what is the purpose of this ?